### ISSUE TITLE
Code formatting is not working with quarkus-java11 plugin

### ISSUE THREAD

**Issue by sunix** [eclipse-che/che#17560]
### Describe the bug
Tried to format Java code with 
- 7.16.2 (che.openshift.io)
- quarkus-java11 plugin
  ```yaml
    - id: redhat/quarkus-java11/latest
      type: chePlugin
  ```

Code completion works fine.

**Comment by vitaliy-guliy** [comment]
Che-Theia does not want to format file when several formatters are registered for the language.

Quarkus11 plugin adds folowing extensions to Che-Theia
```
- https://download.jboss.org/jbosstools/vscode/3rdparty/vscode-java-debug/vscode-java-debug-0.26.0.vsix
- https://download.jboss.org/jbosstools/static/jdt.ls/stable/java-0.63.0-2222.vsix
- https://download.jboss.org/jbosstools/vscode/stable/vscode-quarkus/vscode-quarkus-1.5.0-324.vsix
```
Java and vscode-quarkus extensions register formatters for Java language. Command to format the document is displayed and enabled in the context menu for java file, but it does nothing.

![Screenshot from 2020-08-13 17-28-55](https://user-images.githubusercontent.com/1655894/90150924-d52fde00-dd8e-11ea-9c5d-8bde0e158d7d.png)

I recorded a small video how VS Code behaves when I tried to run it with those three extensios https://www.youtube.com/watch?v=ekpMNaLvCWo

Corresponding issue in Theia upstream https://github.com/eclipse-theia/theia/issues/8354

**Comment by sunix** [comment]
but shouldn't it just pick the last one?

**Comment by ericwill** [comment]
> but shouldn't it just pick the last one?

When possible, we should give the choice to the user. Also this functionality is bound to come up again, so having it in Theia makes sense and is worth the effort.

**Comment by benoitf** [comment]
it seems Quarkus extension shouldn't do formatting anymore https://github.com/redhat-developer/vscode-quarkus/issues/166 https://github.com/redhat-developer/quarkus-ls/pull/332

**Comment by ericwill** [comment]
> it seems Quarkus extension shouldn't do formatting anymore [redhat-developer/vscode-quarkus#166](https://github.com/redhat-developer/vscode-quarkus/issues/166) [redhat-developer/quarkus-ls#332](https://github.com/redhat-developer/quarkus-ls/pull/332)

@vitaliy-guliy can you check the latest `vscode-quarkus` plugin (I believe it's 1.6.0) and see if it removes the formatter issue?

**Comment by vitaliy-guliy** [comment]
There is a PR https://github.com/eclipse/che-plugin-registry/pull/575 with quarkus 1.6.0
After running the workspace and opening a java file, a message 'The workspace contains Java projects' is appeared and user is proposed to import them.
This is a bit strange, but is not a big problem. After clicking 'Yes', the language server is being running and it's become possible to format the document.

![Screenshot from 2020-08-18 14-43-50](https://user-images.githubusercontent.com/1655894/90509823-9b7e1f00-e162-11ea-9028-b0546100d4bd.png)

**Comment by benoitf** [comment]
we should have as default setting for new vscode java versions:

```
java.project.importOnFirstTimeStartup = automatic
```

**Comment by vitaliy-guliy** [comment]
@benoitf thanks. The devfile in my previous comment is updated.

**Comment by benoitf** [comment]
@vitaliy-guliy `java.project.importOnFirstTimeStartup: 'true'` , `true` looks like different than `automatic` ?

**Comment by vitaliy-guliy** [comment]
@benoitf  You won't believe it, but I set it to 'true' automatically. And as it turned out, it also works.

**Comment by benoitf** [comment]
@vitaliy-guliy yes it works due to implementation details
https://github.com/redhat-developer/vscode-java/blob/e541b1a8dc2cf2127ef9bbe34c1474d24d8fd65f/src/extension.ts#L309

but it's not really what is expected

> java.project.importOnFirstTimeStartup: Specifies whether to import the Java projects, when opening the folder in Hybrid mode for the first time. Supported values are disabled (never imports), interactive (asks to import or not), automatic (always imports). Default to interactive.

**Comment by ericwill** [comment]
I think we can close this, as we have vscode-quarkus 1.6.0 which does not contribute any formatters anymore. The theia issue is separate and can be handled as such.

**Comment by sunix** [comment]
so fixed with https://github.com/eclipse/che-plugin-registry/pull/575 and https://github.com/redhat-developer/quarkus-ls/pull/332

**Comment by ericwill** [comment]
> so fixed with [eclipse/che-plugin-registry#575](https://github.com/eclipse/che-plugin-registry/pull/575) and [redhat-developer/quarkus-ls#332](https://github.com/redhat-developer/quarkus-ls/pull/332)

Correct
